local_planner/src/nodes/local_planner.cpp

```diff
@@ -167,7 +166,7 @@ void LocalPlanner::determineStrategy() {
   }
 
   if (!reach_altitude_) {
-    starting_height_ = std::max(goal_.z() - 0.5f, take_off_pose_.z() + 1.0f);
```

Comment:
In the case you describe: current position > (goal.z - 0.5) and planner is restarted -> takeoff_pose >(goal.z - 0.5)

Then the starting height will be set to takeoff_pose +1.0 which is higher up than the goal. I agree that this is not ideal but I do not see a critical problem in that. The drone will rise 1m from the current position to the starting height and will start its mission. During the travel to the next wp it will loose the 1m again. Or that is what I would expect to happen. Do you see something different?

Reply:
it can happen that the vehicle never reaches the `starting_height` because `position.z > goal.z` therefore the waypoint generator produces a setpoint to lower the vehicle altitude (https://github.com/PX4/avoidance/blob/master/local_planner/src/nodes/waypoint_generator.cpp#L174-L177).

This is the case I am seeing in simulation.

Reply:
Ah ok , true that makes sense. But then another solution would be to swap the goal.z with starting_height in those lines. Such that it will send waypoints up until the starting height is reached, right? I don't know which is the better solution. I'm just a tiny bit worried that removing the sanity check in starting height will break the offboard workflow.

Reply:
Found another issue: the goal position gets overwritten [here with the current position](https://github.com/PX4/avoidance/blob/master/local_planner/src/nodes/waypoint_generator.cpp#L168-L169).  Since when restarting the node the goal.z and position.z are closer and `speed_` the vehicle is stuck at current position because of https://github.com/PX4/avoidance/blob/master/local_planner/src/nodes/waypoint_generator.cpp#L259-L260